src/item_pocket.cpp

```diff
@@ -2047,6 +2063,45 @@ void item_pocket::favorite_settings::clear_category( const item_category_id &id
     category_whitelist.erase( id );
 }
 
+/**
+ * Check to see if the given container is accepted by this pocket based on player configured rules.
+ * In order for a container to be accepted all items contained inside must be accepted.
+ *
+ * @param container item container to check.
+ * @return true if the container is accepted by this pocket, false otherwise.
+ * @see item_pocket::favorite_settings::accepts_item(const item&)
+ */
+bool item_pocket::favorite_settings::accepts_container( const item &container ) const
+{
+    std::list<const item *>::const_iterator iter;
+    const std::list<const item *> items = container.all_items_top();
+    for( iter = items.begin(); iter != items.end(); ++iter ) {
+        const item *it = *iter;
+        if( !accepts_item( *it ) ) {
+            return false;
+        }
+    }
+    return true;
```

Comment:
```cpp
const std::list<const item *> items = container.all_items_top();
std::all_of( items.begin(), items.end(), []( const item *i ) {
    return accepts_item( i );
} );
```

Reply:
Code refactored in https://github.com/CleverRaven/Cataclysm-DDA/pull/54812/commits/3a62df009914e6776f1c69073e1f1bedd78be382